Source/JavaScriptCore/builtins/BuiltinExecutables.cpp

```diff
@@ -265,14 +265,22 @@ UnlinkedFunctionExecutable* BuiltinExecutables::createExecutable(VM& vm, const S
     return functionExecutable;
 }
 
-void BuiltinExecutables::finalizeUnconditionally(CollectionScope)
+template<typename Visitor>
+void BuiltinExecutables::visitAggregateImpl(Visitor& visitor)
 {
     for (auto*& unlinkedExecutable : m_unlinkedExecutables) {
-        if (unlinkedExecutable && !m_vm.heap.isMarked(unlinkedExecutable))
-            unlinkedExecutable = nullptr;
+        if (unlinkedExecutable)
+            visitor.appendUnbarriered(unlinkedExecutable);
```

Comment:
What are `visitAggregate` functions used for? And what is `appendUnbarriered` trying to achineve?